python/src/scanner.rs

```diff
@@ -28,6 +30,129 @@ use pyo3::exceptions::PyValueError;
 use crate::reader::LanceReader;
 use crate::RT;
 
+#[pyclass]
+pub struct LanceScanStats {
+    inner: ScannerStats,
+}
+
+impl LanceScanStats {
+    pub fn new(inner: ScannerStats) -> Self {
+        Self { inner }
+    }
+}
+
+#[pymethods]
+impl LanceScanStats {
+    /// The start of the scan
+    ///
+    /// This is when the stream is constructed, not when it is first consumed.
+    ///
+    /// Returned as milliseconds since the UNIX epoch
+    #[getter]
+    fn start(&self) -> PyResult<u64> {
+        Ok(self
+            .inner
+            .start
+            .duration_since(std::time::UNIX_EPOCH)
+            .unwrap()
+            .as_millis()
+            .try_into()
+            .unwrap())
+    }
```

Comment:
If you're willing to make `start` and `end` `datetime.datetime` objects instead of ints, the [PyO3 conversion table](https://pyo3.rs/v0.23.4/conversions/tables.html) suggests you should just be able to do:
```suggestion
    #[getter]
    fn start(&self) -> SystemTime {
        self.inner.start.clone()
    }
```